Remotion/Web/Core/Res/Themes/NovaGray/HTML/StyleUtility.js

```diff
@@ -26,31 +26,31 @@ StyleUtility.CreateBorderSpans = function (selector)
 
 StyleUtility.AddBrowserSwitch = function ()
 {
-  var browser;
-  if (navigator.appVersion.indexOf ('Edge') !== -1)
+  function searchBrowser(brands, search)
   {
-    browser = 'edge';
+    return brands.some(c => c.brand === search);
   }
-  else if (navigator.appVersion.indexOf ('Chrome') !== -1)
-  {
-    browser = 'blink';
-  }
-  else if (navigator.appVersion.indexOf ('OPR') !== -1)
-  {
-    browser = 'blink';
-  }
-  else if (navigator.appVersion.indexOf('WebKit') !== -1)
-  {
-    browser = 'webkit';
-  }
-  else if (navigator.appName === 'Netscape')
+
+  let browser;
+  if (navigator.userAgentData)
   {
-    browser = 'mozilla';
+    const brands = navigator.userAgentData.brands;
+    if (searchBrowser(brands, 'Microsoft Edge') || searchBrowser(brands, 'Google Chrome') || searchBrowser(brands, 'Opera'))
+      browser = 'blink';
   }
-  else
+
+  if(!browser)
   {
```

Comment:
Semantics of the feature detection: as far as I understand the situation, it's either the feature is avaible or it is not. the feature detection should therefor do an if/else based on the availability of the API, not based on whether a match was found. The fallback is only needed when the API is not available.

Reply:
Yes, but if a browser adds the feature we won't detect them correctly and mistakenly identify it as `browserUnknown`. To make sure that does not happen we need to use the user agent detection as fallback.

Reply:
oh, I get it. Okay, that's true. @KuJo7 sorry, I didn't realize this detail during the review. Maybe we should add a code comment for the intent to future-proof this?